x-pack/plugins/apm/server/lib/services/get_service_environments.ts

```diff
@@ -0,0 +1,56 @@
+import { BucketAgg, ESFilter } from 'elasticsearch';
+import { idx } from '@kbn/elastic-idx';
+import {
+  PROCESSOR_EVENT
+} from '../../../common/elasticsearch_fieldnames';
+import { PromiseReturnType } from '../../../typings/common';
+import { rangeFilter } from '../helpers/range_filter';
+import { Setup } from '../helpers/setup_request';
+
+export type ServiceEnvironmentsAPIResponse = PromiseReturnType<typeof getServiceEnvironments>;
+export async function getServiceEnvironments(setup: Setup) {
+  const { start, end, esFilterQuery, client, config } = setup;
+
+  const filter: ESFilter[] = [
+    { terms: { [PROCESSOR_EVENT]: ['transaction', 'error', 'metric'] } },
+    { range: rangeFilter(start, end) }
+  ];
+
+  if (esFilterQuery) {
+    filter.push(esFilterQuery);
+  }
+
+  const params = {
+    index: [
+      config.get<string>('apm_oss.metricsIndices'),
+      config.get<string>('apm_oss.errorIndices'),
+      config.get<string>('apm_oss.transactionIndices')
+    ],
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter
+        }
+      },
+      aggs: {
+        environments: {
+          terms: { field: 'service.environment' }
```

Comment:
Let's set an explicit bucket size. I think it defaults to 100. Not sure what we want to set it at.